lib/deck.rb

```diff
@@ -0,0 +1,35 @@
+require 'pry'
+require './lib/card'
+class Deck
+  attr_accessor :cards, :cards_in_category
+
+  def initialize(cards)
+    @cards = cards
+  end
+
+  def count
+    return cards.length
+  end
+
+  def cards_in_category(category)
+    @cards_in_category = []
+    i = 0
+    while i < @cards.count
+      if @cards[i].category == category
+
+           @cards_in_category << @cards[i]
+      end
+      i += 1
+    end
+    return @cards_in_category
+    #unsuccessful! Ask why
+    #try passing it a specific index
+    # @cards.each_with_index do |card, index|
+    #   if @cards[index].category == @category
+    #     @cards_in_category << @card[index]
+    #   end
+    # end
+    # return @cards_in_category
+
+  end
```

Comment:
This works! But we can definitely slim it up a bit, and uncomplicate (is that a word?) some of your logic: 

```ruby
  def cards_in_category(category)
    cards_in_category = []
    @ cards.each do |card|
      if card.category == category
           cards_in_category << card
      end
    end
    cards_in_category
```


There's a few things I did here .. instead of querying the cards array with an incremented index counter, I'm iterating through the cards collection since I know I want to look at every single one of them. Each-ing through the cards collection allows me to get to each card element in the array without having to manually look up the next index in the collection. 
I also am storing the cards_in_category array in a local variable instead of an instance variable. I'm doing this because I only need access to this variable within THIS method. We use instance variables (variables with the @ sign) when we want to access those values outside of the method that it was declared in.